File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    used = set()
    N = len(words)
    cnt = 0
    
    for i in range(N):
        if words[i] in used:
            answer.append(i%n+1)
            if (i+1)%n > 0:
                answer.append(((i+1)//n + 1))
            else:
                answer.append((i+1)//n)
            
            break
        else:
            if i != 0:
                if words[i][0] != words[i-1][-1]:
                    answer.append(i%n+1)
                    if (i+1)%n > 0:
                        answer.append(((i+1)//n) + 1)
                    else:
                        answer.append((i+1)//n)
                    break
            used.add(words[i])

    if not answer:
        answer.append(0)
        answer.append(0)

    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.Set;
import java.util.HashSet;
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {};
        int N = words.length;
        Set<String> used = new HashSet<>();
        boolean possible = true;
        
        for (int i = 0; i < N; i++) {
            if (used.contains(words[i])) {
                possible = false;
                if ((i+1) % n > 0) {
                    answer = new int[] {(i%n) + 1, (i+1) / n + 1};
                } else {
                    answer = new int[] {(i%n) + 1, (i+1) / n};
                }
                break;
            } else {
                if (i != 0) {
                     if (words[i].charAt(0) != words[i-1].charAt(words[i-1].length() - 1)) {
                        possible = false;
                        if ((i+1) % n > 0) {
                            answer = new int[] {(i%n) + 1, (i+1) / n + 1};
                        } else {
                            answer = new int[] {(i%n) + 1, (i+1) / n};
                        }
                        break;
                     }
                }
                used.add(words[i]);
            }  
        }
        if (possible) {
            answer = new int[] {0, 0};
        }
        return answer;
    }
}